client/extensions/woocommerce/index.js

```diff
@@ -151,15 +155,23 @@ function addStorePage( storePage, storeNavigation ) {
 	} );
 }
 
-function createStoreNavigation( context, next ) {
-	const state = context.store.getState();
-	const selectedSite = getSelectedSite( state );
+function mapStateToProps( state ) {
+	let siteId = getSelectedSiteId( state );
+	if ( ! siteId && ( 1 === getCurrentUserSiteCount( state ) ) ) {
+		siteId = getPrimarySiteId( state );
+	}
+	const site = getSite( state, siteId );
 
+	return {
+		site
+	};
+}
+
+function createStoreNavigation( context, next ) {
 	renderWithReduxStore(
 		React.createElement(
-			StoreSidebar, {
+			connect( mapStateToProps )( StoreSidebar ), {
```

Comment:
On a closer look here, I think I'd recommend that the `connect()` statement be moved to `store-sidebar/index.js` as that's the way we normally use it. We just wrap it in `connect()` before we export.